src/pretix/presale/forms/renderers.py

```diff
@@ -1,5 +1,38 @@
 from bootstrap3.renderers import FieldRenderer
+from bootstrap3.text import text_value
 from bootstrap3.utils import add_css_class
+from django.forms import CheckboxInput
+from django.forms.utils import flatatt
+from django.utils.html import format_html
+from django.utils.safestring import mark_safe
+from django.utils.translation import pgettext
+
+
+def render_label(content, label_for=None, label_class=None, label_title='', optional=False):
+    """
+    Render a label with content
+    """
+    attrs = {}
+    if label_for:
+        attrs['for'] = label_for
+    if label_class:
+        attrs['class'] = label_class
+    if label_title:
+        attrs['title'] = label_title
+
+    if text_value(content) == '&#160;':
+        # Empty label, e.g. checkbox
+        attrs.setdefault('class', '')
+        attrs['class'] += ' label-empty'
```

Comment:
The checkbox case is probably one we need to handle differently, a label of just "required" probably doesn't make a lot of sense?

Reply:
No, but an empty label is wrong anyways. Will look into this. Do you have any examples where I can find such a checkbox with an empty label?

Reply:
The checkbox will have another, non empty label.

You can probably construct this in the frontend by just adding a question of type Yes/No.

Reply:
Ah, I see – I had a Yes/no-Question, but it was not required. I will look into this.

Reply:
Fixed. I added "required" to the  checkbox-label (when inside `put_inside_label`) as well as removed it from the overall-label for checkboxes, when it is empty (which it usually is)